src/Shared/Debugging/DebugUtils.cs

```diff
@@ -34,10 +34,15 @@ internal static void SetDebugPath()
 
             if (Traits.Instance.DebugEngine)
             {
-                if (!string.IsNullOrWhiteSpace(debugDirectory) && FileUtilities.CanWriteToDirectory(debugDirectory))
+                if (!string.IsNullOrWhiteSpace(debugDirectory) && FileUtilities.CanWriteToDirectory(debugDirectory) && !IsPathInSolutionDirectory(debugDirectory))
                 {
                     // Debug directory is writable; no need for fallbacks
                 }
+                else if (!string.IsNullOrWhiteSpace(debugDirectory) && IsPathInSolutionDirectory(debugDirectory))
```

Comment:
I'd remove this else if branch and add the `&& !IsPathInSolutionDirectory(debugDirectory)` also to the next `else if` so that the code is less duplicated and the semantics is clear: "the specified directories are not available, so we go to temp"